code/game/objects/items/devices/scanners.dm

```diff
@@ -1031,99 +1081,221 @@ GENE SCANNER
+			return
+		if(EXTRAPOLATOR_SPECIAL_HANDLED in diseases)
+			// extrapolator_act did some sort of special behavior, we don't need to do anything further
+			return
+		if(scan)
+			scan(user, target)
+		else
+			extrapolate(user, target)
 	else
 		to_chat(user, "<span class='warning'>The extrapolator has no scanner installed!</span>")
 
-/obj/item/extrapolator/proc/scan(atom/AM, var/list/diseases = list(), mob/user)
-	to_chat(user, "<span class='notice'><b>[src] detects the following diseases:</b></span>")
-	for(var/datum/disease/D in diseases)
-		if(istype(D, /datum/disease/advance))
-			var/datum/disease/advance/A = D
-			if(A.stealth >= (2 + scanner.rating)) //the extrapolator can detect diseases of higher stealth than a normal scanner
-				continue
-			if(A.dormant)
-				to_chat(user, "<span class='info'><font color='A19D9C'><b>[A.name]</b>, dormant virus</font></span>")
-				to_chat(user, "<span class='info'><font color='BAB9B9'><b>[A] has the following symptoms:</b></font></span>")
-				for(var/datum/symptom/S in A.symptoms)
-					to_chat(user, "<span class='info'><font color='BAB9B9'>[S.name]</font></span>")
-			else
-				to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</font></span>")
-				to_chat(user, "<span class='info'><b>[A] has the following symptoms:</b></span>")
-				for(var/datum/symptom/S in A.symptoms)
-					to_chat(user, "<span class='info'>[S.name]</span>")
+/obj/item/extrapolator/proc/find_valid_targets(mob/living/user, atom/target)
+	. = list()
+	var/turf/target_turf = get_turf(target)
+	if(!target_turf)
+		return
+	for(var/atom/target_to_try in target_turf.contents - target)
+		if(length(target_to_try.extrapolator_act(user, src, dry_run = TRUE)))
+			. += target_to_try
+
+/**
+ * Gets the color to display in chat based on a severity level.
+ */
+/obj/item/extrapolator/proc/get_danger_color(value)
+	if(istype(value, /datum/disease/advance))
+		var/datum/disease/advance/advance_disease = value
+		if(advance_disease.dormant)
+			return COLOR_SILVER
+		value = advance_disease.severity
+	else if(istype(value, /datum/symptom))
+		var/datum/symptom/symptom = value
+		value = symptom.severity
+	switch(value)
+		if(-INFINITY to -2)
+			return COLOR_LIME
+		if(-1)
+			return COLOR_BLUE_LIGHT
+		if(0)
+			return COLOR_BROWN
+		if(1)
+			return COLOR_PALE_ORANGE
+		if(2)
+			return COLOR_YELLOW
+		if(3)
+			return COLOR_ORANGE
+		if(4)
+			return COLOR_RED_LIGHT
+		if(5)
+			return COLOR_DARK_RED
+		if(6 to INFINITY)
+			return COLOR_MAROON
 		else
-			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</font></span>")
-
-/obj/item/extrapolator/proc/extrapolate(atom/AM, var/list/diseases = list(), mob/user, isolate = FALSE, timer = 100)
-	var/list/advancediseases = list()
-	var/list/symptoms = list()
-	if(using)
-		to_chat(user, "<span class='warning'>The extrapolator is already in use.</span>")
+			return COLOR_SILVER
```

Comment:
avoid using direct color defines. You should use class name css colors, and add acceptable color to dark/light chat themes.

Reply:
... Do you know any spans which would fit well here?

Reply:
Meh, I just removed the colors.